spinn_front_end_common/interface/interface_functions/application_finisher.py

```diff
@@ -62,16 +62,16 @@ def application_finisher():
                 f"{total_processors} processors went into an error state "
                 "when shutting down")
 
-        successful_cores_finished = txrx.get_cpu_infos(
-            all_core_subsets, CPUState.FINISHED, True)
+        successful_cores_finished = set(txrx.get_cores_in_states(
+            all_core_subsets, CPUState.FINISHED))
 
         txrx.send_signal(app_id, Signal.SYNC0)
         txrx.send_signal(app_id, Signal.SYNC1)
 
         for core_subset in all_core_subsets:
             for processor in core_subset.processor_ids:
-                if not successful_cores_finished.is_core(
-                        core_subset.x, core_subset.y, processor):
+                if (core_subset.x, core_subset.y, processor) not in \
+                        successful_cores_finished:
```

Comment:
Is that change necessary?  It seems that the function get_cores_in_states returns the same information as get_cpu_infos, but it has now been put in a set instead of using the built-in set...

Reply:
The difference is that get_cpu_infos promises to return a spinnman.model.CPUInfos while get_cores_(not_)in_states only contracts to return an iterable(int, int, int) xyp.   This gives Spinn2 the freedom the have a different "CPUInfos" model